core/domain/base_model_validators_test.py

```diff
@@ -190,3 +212,84 @@ def test_validate_deleted_reports_error_for_old_deleted_model(self):
                 ]
             }
         )
+
+    def test_external_model_fetcher_with_user_settings(self):
+        with self.assertRaisesRegexp(
+            Exception,
+            'When fetching instances of UserSettingsModel, please use ' +
+            'UserSettingsModelFetcherDetails instead of ' +
+            'ExternalModelFetcherDetails'):
+            base_model_validators.ExternalModelFetcherDetails(
+                'committer_ids', user_models.UserSettingsModel,
+                [
+                    feconf.MIGRATION_BOT_USER_ID, 'User-1',
+                    self.PSEUDONYMOUS_ID
+                ]
+            )
+
+    def test_may_contain_system_users(self):
+        user_settings_model = (
+            base_model_validators.UserSettingsModelFetcherDetails(
+                'committer_ids',
+                [feconf.MIGRATION_BOT_USER_ID, 'User-1'],
+                may_contain_system_ids=True,
+                may_contain_pseudonymous_ids=False
+            ))
+
+        self.assertItemsEqual(
+            user_settings_model.model_ids, ['User-1'])
+
+    def test_error_raised_if_model_ids_contain_system_ids(self):
+        with self.assertRaisesRegexp(
+            utils.ValidationError,
+            'The field committer_ids should not contain system ids but it '
+            'contains system ids'):
+            base_model_validators.UserSettingsModelFetcherDetails(
+                'committer_ids', [feconf.MIGRATION_BOT_USER_ID, 'User-1'],
+                may_contain_system_ids=False,
+                may_contain_pseudonymous_ids=False
+            )
+
+    def test_may_contain_pseudonymous_users(self):
+        user_settings_model = (
+            base_model_validators.UserSettingsModelFetcherDetails(
+                'committer_ids', ['User-1', self.PSEUDONYMOUS_ID],
+                may_contain_system_ids=False,
+                may_contain_pseudonymous_ids=True
+            ))
+
+        self.assertItemsEqual(
+            user_settings_model.model_ids, ['User-1'])
+
+    def test_error_raised_if_model_ids_contain_pseudonymous_ids(self):
+        with self.assertRaisesRegexp(
+            utils.ValidationError,
+            'The field committer_ids should not contain pseudonymous ids but '
+            'it contains pseudonymous ids'):
+            base_model_validators.UserSettingsModelFetcherDetails(
+                'committer_ids', [self.PSEUDONYMOUS_ID, 'User-1'],
+                may_contain_system_ids=False,
+                may_contain_pseudonymous_ids=False
+            )
+
+    def test_error_raised_when_fetching_external_model_with_system_ids(self):
+        model = MockCommitLogEntryModel(
+            id='12345', user_id=feconf.MIGRATION_BOT_USER_ID)
+        mock_validator = MockCommitLogEntryModelValidator()
+        mock_validator.errors.clear()
+        mock_validator._fetch_field_name_to_external_model_references(model) # pylint: disable=protected-access
```

Comment:
Try to limit the usage of `disable=protected-access`, can we call some public method instead?

Reply:
Done.